modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -289,7 +295,18 @@ def run_host(ip)
         smb_login
 
         begin
-          shares = simple.client.net_share_enum_all(ip)
+          # Return all shares if `Share` option has not been set
+          if datastore['Share'].nil?
+            shares = simple.client.net_share_enum_all(ip)
+          else
+            # Return specific share if the `Share` option has been set
+            simple.client.net_share_enum_all(ip).each { |share| shares = [share] if share[:name] == datastore['Share'] }
+            # Return an error and all shares if `Share` option has been set but no matches were found
+            if shares.empty?
+              print_error("No shares match #{datastore['Share']} - returning all shares instead")
+              shares = simple.client.net_share_enum_all(ip)
+            end
```

Comment:
I'm in favor of failing here immediately if the share doesn't exist, especially in the scenario where the user has specified ShowFiles set to true it will output a lot of information to the screen that the user didn't request.

Reply:
Yea I was unsure whether to just fail on this scenario or whether to default to the modules standard functionality. Was thinking the users workflow would be to immediately rerun the module without the option to see if they had misconfigured and so on. 

I swapped over to just failing now as I agree that it's a bit noisy and forces the user to scroll through all the shares to see what they even failed on.

With the new changes I have push it will now attempt to connect to the share and if the connect fails it will return an error:
```
Error when trying to enumerate shares - STATUS_BAD_NETWORK_NAME
```

Let me know if there is any issues with that 👍